referenceframe/geometry_proto_test.go

```diff
@@ -0,0 +1,259 @@
+package referenceframe
+
+import (
+	"math"
+	"testing"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+	commonpb "go.viam.com/api/common/v1"
+	"go.viam.com/test"
+
+	"go.viam.com/rdk/pointcloud"
+	"go.viam.com/rdk/spatialmath"
+)
+
+func TestGeometryProtobufRoundTrip(t *testing.T) {
+	deg45 := math.Pi / 4
+	testCases := []struct {
+		name     string
+		geometry spatialmath.Geometry
+	}{
+		{"box", spatialmath.MakeTestBox(&spatialmath.EulerAngles{0, 0, deg45}, r3.Vector{0, 0, 0}, r3.Vector{2, 2, 2}, "box")},
+		{"sphere", spatialmath.MakeTestSphere(r3.Vector{3, 4, 5}, 10, "sphere")},
+		{"point", spatialmath.MakeTestPoint(r3.Vector{3, 4, 5}, "point")},
+		{"capsule", spatialmath.MakeTestCapsule(&spatialmath.EulerAngles{0, 0, deg45}, r3.Vector{1, 2, 3}, 5, 20, "capsule")},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			proto := testCase.geometry.ToProtobuf()
+			test.That(t, proto, test.ShouldNotBeNil)
+			test.That(t, proto.Center, test.ShouldNotBeNil)
+			test.That(t, proto.Label, test.ShouldEqual, testCase.name)
+
+			newGeometry, err := NewGeometryFromProto(proto)
+			test.That(t, err, test.ShouldBeNil)
+			test.That(t, newGeometry, test.ShouldNotBeNil)
+			test.That(t, testCase.geometry.Label(), test.ShouldEqual, newGeometry.Label())
+			test.That(t, spatialmath.GeometriesAlmostEqual(testCase.geometry, newGeometry), test.ShouldBeTrue)
+		})
+	}
+}
+
+func TestPointCloudProtobufRoundTrip(t *testing.T) {
+	pc := pointcloud.MakeTestPointCloud("pointcloud")
```

Comment:
Yeah actually can we just remove this function completely?  We don't want to be invoking it and then having the test later check for the hard coded contents of it